soroban-env-host/src/test/token.rs

```diff
@@ -224,24 +241,67 @@ impl TokenTest {
     fn convert_bytes(&self, bytes: &[u8]) -> Bytes {
         Bytes::try_from_val(&self.host, self.host.bytes_new_from_slice(bytes).unwrap()).unwrap()
     }
+
+    fn run_from_contract<T, F>(
+        &self,
+        contract_id_bytes: &BytesN<32>,
+        f: F,
+    ) -> Result<RawVal, HostError>
+    where
+        T: Into<RawVal>,
+        F: FnOnce() -> Result<T, HostError>,
+    {
+        self.host.with_frame(
+            Frame::TestContract(
+                Hash(contract_id_bytes.to_array().unwrap()),
+                Symbol::from_str("foo"),
+            ),
+            || {
+                let res = f();
+                match res {
+                    Ok(v) => Ok(v.into()),
+                    Err(e) => Err(e),
+                }
+            },
+        )
+    }
+
+    fn run_from_account<T, F>(&self, account_id: AccountId, f: F) -> Result<RawVal, HostError>
+    where
+        T: Into<RawVal>,
+        F: FnOnce() -> Result<T, HostError>,
+    {
+        self.host.set_source_account(account_id);
+        self.host
+            .with_frame(Frame::HostFunction(HostFunction::InvokeContract), || {
+                let res = f();
+                match res {
+                    Ok(v) => Ok(v.into()),
+                    Err(e) => Err(e),
+                }
+            })
+    }
```

Comment:
@dmkozh This is a really neat idea! 👏🏻  I think we should add the same thing to the `Env` in `testutils` in the SDK, as there are some situations where in a test we want to be able to access something _as_ a contract, and we currently have no way of doing that.